[Index Management] Add support for enrich policies

sabarasaba opened the pull request:
This PR is a feature branch that adds support for managing, executing and deleting enriched policies in index management.

## Release Notes
This feature adds support for enriched policies in index management. Providing the UX to create, as well as executing and deleting them.

## Note to reviewers
This is a feature branch that contains few other smaller PR's that were already reviewed and tested individually. The list of PR's that are included in this feature branch are:

* https://github.com/elastic/kibana/pull/163159
* https://github.com/elastic/kibana/pull/164130
* https://github.com/elastic/kibana/pull/164131
* https://github.com/elastic/kibana/pull/166144
* https://github.com/elastic/kibana/pull/166047
* https://github.com/elastic/kibana/pull/166361

sabarasaba commented:
@elasticmachine merge upstream

sabarasaba commented:
@elasticmachine merge upstream

sabarasaba commented:
@elasticmachine merge upstream

sabarasaba commented:
@elasticmachine merge upstream

sabarasaba commented:
@elasticmachine merge upstream

elasticmachine commented:
Pinging @elastic/platform-deployment-management (Team:Deployment Management)

alisonelizabeth reviewed (approved):
Awesome work @sabarasaba! 👏 

I ran through the [example scenarios](https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest-enriching-data.html) for each policy type and everything worked as expected.

I did not do a thorough code review as the code has been reviewed already, however, I did call out a couple things I noticed while taking a quick glance through.

Also, two minor UX issues I noticed:
1. Do you think "Source indices" should come before "Match field" to align with the order of the fields in the form?

<img width="534" alt="Screenshot 2023-09-19 at 2 53 12 PM" src="https://github.com/elastic/kibana/assets/5226211/77997d68-ef5a-4976-a90c-691415a5bc63">

2. Should a user be able to hover over the ellipsis and see the overflow values? 
<img width="296" alt="Screenshot 2023-09-19 at 2 56 00 PM" src="https://github.com/elastic/kibana/assets/5226211/e4161a92-a3f2-4d2d-9f38-5b37706dea41">

sabarasaba commented on x-pack/plugins/index_management/public/application/sections/enrich_policy_create/steps/fields/indices_selector.tsx:
```diff
@@ -0,0 +1,124 @@
+    [setIsIndiciesLoading, setIndexOptions]
+  );
+
+  // Load first 10 indices on mount so that the ComboBox has some options
```
It is correct, as the api returns the only the first 10 matching indices by default, I've changed the comment so its less confussing!

sabarasaba commented on x-pack/plugins/index_management/public/application/sections/home/enrich_policies_list/confirm_modals/delete_policy_modal.tsx:
```diff
@@ -0,0 +1,104 @@
+      <p>
+        <FormattedMessage
+          id="xpack.idxMgmt.enrichPolicies.deleteModal.bodyCopy"
+          defaultMessage="You are about to delete the enrich policy {policy}. This action is irreverisble."
```
woaaaa nice catch!!

sabarasaba commented:
Thanks for checking it out @alisonelizabeth! I've addressed your comments with https://github.com/elastic/kibana/pull/166790, but for the tooltip on the table cell I'm not sure its a good idea because if they have very long fields or just many of them reading them on on a small tooltip isn't the best experience. I think its better for them to simply open the flyout, wdy?

alisonelizabeth commented:
> Thanks for checking it out @alisonelizabeth! I've addressed your comments with https://github.com/elastic/kibana/pull/166790, but for the tooltip on the table cell I'm not sure its a good idea because if they have very long fields or just many of them reading them on on a small tooltip isn't the best experience. I think its better for them to simply open the flyout, wdy?

Thanks for addressing my feedback! I'm fine to leave the table cell as-is for now.

sabarasaba commented:
/oblt-deploy
